test/gtest/uct/ib/test_ib.cc

```diff
@@ -866,29 +867,37 @@ class test_uct_event_ib : public test_uct_ib {
         }
     }
 
-    void check_send_cq(uct_iface_t *iface, size_t val) {
-        uct_ib_iface_t *ib_iface = ucs_derived_of(iface, uct_ib_iface_t);
-        struct ibv_cq  *send_cq = ib_iface->cq[UCT_IB_DIR_TX];
-
-        if (val != send_cq->comp_events_completed) {
-            uint32_t completed_evt = send_cq->comp_events_completed;
-            /* need this call to acknowledge the completion to prevent iface dtor hung*/
-            ibv_ack_cq_events(ib_iface->cq[UCT_IB_DIR_TX], 1);
-            UCS_TEST_ABORT("send_cq->comp_events_completed have to be 1 but the value "
-                           << completed_evt);
+    size_t get_comp_events(uct_iface_t *iface, uct_ib_dir_t dir)
+    {
+        uct_ib_iface_t *ib_iface;
+#if IBV_HW_TM
+        uct_rc_mlx5_iface_common_t *rc_mlx5_iface;
+        if (has_transport("dc_mlx5") || has_transport("rc_mlx5")) {
+            rc_mlx5_iface = ucs_derived_of(iface, uct_rc_mlx5_iface_common_t);
+            return rc_mlx5_iface->cq[dir].cq_sn;
         }
+#endif
+        ib_iface = ucs_derived_of(iface, uct_ib_iface_t);
+        return ib_iface->cq[dir]->comp_events_completed;
     }
 
-    void check_recv_cq(uct_iface_t *iface, size_t val) {
+    void ack_cq(uct_iface_t *iface, uct_ib_dir_t dir)
+    {
         uct_ib_iface_t *ib_iface = ucs_derived_of(iface, uct_ib_iface_t);
-        struct ibv_cq  *recv_cq = ib_iface->cq[UCT_IB_DIR_RX];
-
-        if (val != recv_cq->comp_events_completed) {
-            uint32_t completed_evt = recv_cq->comp_events_completed;
-            /* need this call to acknowledge the completion to prevent iface dtor hung*/
-            ibv_ack_cq_events(ib_iface->cq[UCT_IB_DIR_RX], 1);
-            UCS_TEST_ABORT("recv_cq->comp_events_completed have to be 1 but the value "
-                           << completed_evt);
+
+        ibv_ack_cq_events(ib_iface->cq[dir], 1);
+    }
+
+    void check_cq(uct_iface_t *iface, uct_ib_dir_t dir, size_t val)
```

Comment:
I know it's not "your code", but I'd replace `val` with `value` (abbreviations are bad and confusing. Is "val" a "value" or "valid"?).

If we could I'd also change `dir` (direction? directory? direct?) but it has the `uct_ib_dir_t` type so we'll leave that for another time :)